### PR TITLE
update styling according to Google styleguide

### PR DESCRIPTION
**Pull request by ankit-m**
This updates curly braces and indentation styling according to [Google style guide](https://google.github.io/styleguide/javascriptguide.xml?showone=Code_formatting#Code_formatting) in all the js files in source.

>Because of implicit semicolon insertion, always start your curly braces on the same line as whatever they're opening. 

``` 
if (something) {
  // ...
} else {
  // ...
}
```

> Multiline array initializers and object initializers are indented 2 spaces

It also fixes missing semicolons and replaces `!=` with `!==` wherever required i.e. comparing with zeroes and null. 

Improves javascript style #60

### CODE REVIEW

File: Source/options.js [hunk 1]
```diff
@@ -301,7 +315,9 @@ $(function() {
         patterns[0][3] = true;
         activatePattern(0, patterns);
       } else {
-        saveBulk({'savedPatterns': JSON.stringify(patterns)}, 'Deleted Pattern');
+        saveBulk({
+          'savedPatterns': JSON.stringify(patterns)
+        }, 'Deleted Pattern');
```
Comment (rohankatyal29) [inline Source/options.js]: Shouldn't the parameters of a function be on the same line?

Comment (ankit-m) [inline Source/options.js]: The function call line is a little. So breaking it this way adds more readability.

File: Source/MindTheWord.js [hunk 2]
```diff
@@ -120,14 +121,14 @@ function length(obj) {
...
   var i,
-      all,
-      shortest,
-      nShortest,
-      n,
-      len,
-      ret = [],
-      obj = {},
-      nOthers;
+    all,
```
Comment (Ceilican) [inline Source/MindTheWord.js]: In this case, I actually prefer when all the vars are aligned to each other.  Does the style guide say anything specific about cases like this?

**Comment by Ceilican** [comment]
@ankit-m , thank you for that! It looks good! Could you make the pull request automatically mergeable, so that I can merge it more easily?

File: Source/MindTheWord.js (hunk 2 above)
Comment (ankit-m) [inline Source/MindTheWord.js]: It doesn't say specifically about variable declaration. But it does say the following for objects.

> Multiline array initializers and object initializers are indented 2 spaces, with the braces on their own line, just like blocks

```
var inset = {
  top: 10,
  right: 20,
  bottom: 15,
  left: 12
};
```

To maintain consistency, I believe we should stick to 2 space indentations for multiple variable declaration.

**Comment by ankit-m** [comment]
@Ceilican I have made the branch automatically mergeable. There were some new commits which caused the conflicts. I have fixed those.

**Comment by Ceilican** [comment]
Thank you!

**Pull request merged**